politeiawww/cmd/politeiawwwcli/commands/edituser.go

```diff
@@ -32,17 +37,60 @@ Response:
 {}`
 
 type EditUserCmd struct {
-	EmailNotifications *uint64 `long:"emailnotifications" optional:"true" description:"Whether to notify via emails"`
+	Args struct {
+		NotifType string `long:"emailnotifications" description:"Email notifications"`
+	}
 }
 
 func (cmd *EditUserCmd) Execute(args []string) error {
+
+	EmailNotifs := map[string]v1.EmailNotificationT{
+		"userproposalchange":        1,
+		"userproposalvotingstarted": 2,
+		"proposalvetted":            4,
+		"proposaledited":            8,
+		"votingstarted":             16,
+		"newproposal":               32,
+		"userauthorizedvote":        64,
+		"commentonproposal":         128,
+		"commentoncomment":          256,
+	}
+
+	var notif v1.EmailNotificationT
+	a, err := strconv.ParseUint(cmd.Args.NotifType, 10, 64)
+	if err == nil {
+		// Numeric action code found
+		notif = v1.EmailNotificationT(a)
+	} else if a, ok := EmailNotifs[cmd.Args.NotifType]; ok {
+		// Human readable action code found
+		notif = a
+
+	} else if a, ok := EmailNotifs[strings.Split(cmd.Args.NotifType, ",")[0]]; ok {
```

Comment:
This is kinda hard to read.  Do you think something like this would read better?

```
else if strings.Contains(cmd.Args.NotifType, ",") {
```

Reply:
This does read better. Using this. Here I was again following `manageuser.go`, which uses this syntax. I did a quick search for this pattern in the rest of the codebase, couldn't find any other uses of it. It looks like `manageuser.go` has a few "style guide" sins, if those exist here.

Reply:
Manage user is a bit different because a list isn't being passed in so checking for a comma doesn't make sense.  

As far as style goes, most of it comes from either [Effective Go](https://golang.org/doc/effective_go.html) or [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments).  The rest is usually personal preference.  [Practical Go](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_comments) is also a good resource.